src/test/java/de/tum/in/www1/artemis/iris/IrisAuthorizationIntegrationTest.java

```diff
@@ -0,0 +1,105 @@
+package de.tum.in.www1.artemis.iris;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.in.www1.artemis.AbstractSpringIntegrationBambooBitbucketJiraTest;
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.iris.IrisMessage;
+import de.tum.in.www1.artemis.domain.iris.IrisMessageContent;
+import de.tum.in.www1.artemis.domain.iris.IrisMessageSender;
+import de.tum.in.www1.artemis.domain.iris.IrisSession;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.service.iris.IrisMessageService;
+import de.tum.in.www1.artemis.service.iris.IrisSessionService;
+
+class IrisAuthorizationIntegrationTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
```

Comment:
I think another name might make more sense here. Something like this:
```suggestion
class IrisSessionActivationIntegrationTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
```

Just so that it is clear that it is only testing the iris activation part and not general authorization.